src/abr/puffer_ttp.cc

```diff
@@ -54,6 +77,48 @@ void PufferTTP::normalize_in_place(size_t i, vector<double> & input)
   }
 }
 
+
+void PufferTTP::calculate_gaussian_values() {
+    double gaussian_coefficient = 1.0/(std_val_*sqrt(2.0*M_PI));
+    int right_pos =(kernel_size_>>1);
+    int left_pos = -right_pos;
+    /*  In our current case, kernel_size=21, so right_pos = 10, left_pos = -10
+        To construct the kernel for bluring, we sample
+        the points whose x-axis is [-10,10] */
+    for (int x = left_pos; x <= right_pos; x++) {
+        double exp_idx = -(x-mean_val_)*(x-mean_val_)/(2.0*std_val_*std_val_);
+        double gaussian_val = gaussian_coefficient * exp(exp_idx);
+        gaussian_kernel_vals_[x-left_pos] = gaussian_val;
```

Comment:
Add whitespaces around each operator (I'll not repeat this comment but please check the other places)